lib/commands/pull.sh

```diff
@@ -13,6 +14,15 @@ pull() {
     return "$EX_SUCCESS"
   fi
 
+  # this tag is exceedingly unlikely to already exist, but if it does, error
+  # out and let the user resolve it
+  (cd "$repo" && git rev-parse --verify "refs/tags/$BEFORE_PULL_TAG" &>/dev/null) && \
+    err "$EX_DATAERR" "Pull marker tag ($BEFORE_PULL_TAG) already exists in $repo. Please resolve this before pulling."
```

Comment:
Actually, couldn't this be just a warning instead of killing everything? Especially when pulling multiple castles there's no reason to ruin the fun for everybody else :-)

Reply:
That should be how it works now, since pull is in a subshell. I'll add another test later.